lib/ruby_lsp/executor.rb

```diff
@@ -77,24 +77,25 @@ def run(request)
         )
       when "textDocument/foldingRange"
         folding_range(uri)
-      when "textDocument/documentLink"
-        document_link(uri)
       when "textDocument/selectionRange"
         selection_range(uri, request.dig(:params, :positions))
-      when "textDocument/documentSymbol"
+      when "textDocument/documentSymbol", "textDocument/documentLink"
         document = @store.get(uri)
 
         # If the response has already been cached by another request, return it
         cached_response = document.cache_get(request[:method])
         return cached_response if cached_response
 
         # Run listeners for the document
-        listener = Requests::DocumentSymbol.new(@message_queue)
-        EventEmitter.new(listener).visit(document.tree) if document.parsed?
-
-        # Store all responses retrieve in this round of visits in the cache. The last one we save must always be the one
-        # related to the request we're receiving since that's the response we want to return to the editor
-        document.cache_set(request[:method], listener.response)
+        document_symbol = Requests::DocumentSymbol.new(uri, @message_queue)
+        document_link = Requests::DocumentLink.new(uri, @message_queue)
+        EventEmitter.new(document_symbol, document_link).visit(document.tree) if document.parsed?
+
+        # Store all responses retrieve in this round of visits in the cache and then return the response for the request
```

Comment:
```suggestion
        # Store all responses retrieved in this round of visits in the cache and then return the response for the request
```
(optional since was already present)

Reply:
This was fixed on the branch this was based on 👍